src/wallet/wallet.cpp

```diff
@@ -1147,6 +1153,50 @@ void CWallet::BlockDisconnected(const std::shared_ptr<const CBlock>& pblock) {
 
 
 
+void CWallet::BlockUntilSyncedToCurrentChain() {
+    const CBlockIndex* initialChainTip;
+    {
+        LOCK(cs_main);
+        initialChainTip = chainActive.Tip();
+    }
+    AssertLockNotHeld(cs_main);
+    AssertLockNotHeld(cs_wallet);
```

Comment:
`boost::thread_specific_ptr<LockStack> lockstack;`
Didn't see that, makes perfect sense then.